platforms/ide/tooling-api/src/main/java/org/gradle/tooling/BuildController.java

```diff
@@ -223,4 +224,17 @@ public interface BuildController {
      * @since 8.6
      */
     void send(Object value);
+
+    /**
+     * Resilient model fetching.
+     *
+     * @since 9.3.0
+     */
+    @Incubating
+    <T extends Model, M, P> FetchModelResult<T, M> fetch(
```

Comment:
❓  Why did you decide to go with a single-target API as opposed to a multiple-target API to support batch requests out of the box?